design/hardware-classification-controller/support-for-new-parameters-hwcc.md

```diff
@@ -0,0 +1,243 @@
+# Add Support for extra hardware parameters in Disk, RAM, CPU and NIC
+ 
+## Table of Contents
+ 
+<!--ts-->
+   * [Hardware Classification](#title)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+      * [Proposal](#proposal)
+         * [Investigation Details](#investigation-details)
+         * [Implementation Details/Notes/Constraints ](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigation](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+      * [References](#references)
+
+## Summary
+ 
+Implement support for more parameters in HWCC which will help to 
+classify a baremetal host on more possible configurations. 
+ 
+## Motivation
+
+We have hardware classification controller which validates and 
+classifies the hosts based on user hardware requirements.
+We would like to add more parameters which helps user to 
+identify the ideal host. This feature helps user to avoid runtime failures
+and also increases the performance for workload deployments.
+
+### Goals
+
+Automatically classify bare metal hosts based on user-provided rules
+in a way that other controllers can use, without having to evaluate
+all of those rules themselves.
+
+We will carry testing of HWCC only on Dell hardware. Testing of HWCC on 
+hardware from other vendors will not be goal of this proposal.
+
+## Proposal
+ 
+We are submitting proposal based on the issue raised in HWCC: 
+[Issue 53](https://github.com/metal3-io/hardware-classification-controller/issues/53)
+
+### Investigation Details
```

Comment:
It seems like the examples in this section are parts of host resources. It would be helpful to have examples of the new API here as well. The larger example farther down is good, but having a small example for each of the new fields will make it clear how we expect someone to use them.

Reply:
Examples for each of new fields are provided in comment section of Issue.
https://github.com/metal3-io/hardware-classification-controller/issues/53#issuecomment-720363210

Reply:
OK, could you copy that information into this document so that someone reviewing it has all of the context in one place?